### ISSUE TITLE
Design - Atomic (step 1+2) asset upload

### ISSUE THREAD

**Issue by scossu** [aic-collections/aicdams-lakeshore#206]
Currently, if a batch upload process is, for whatever reason, abandoned after the files have been uploaded (step 1) but before submitting the metadata (step 2), the assets in that batch are left in an incomplete state, lacking critical metadata (e.g. document type). 

An ideal workflow would be to store the file in a temporary location and ingest it in Fedora only after step 2 is submitted. 

This ticket is to discuss the feasibility, effort and possible implications of an "atomic" batch upload operation as described above. 

N.B. This may be a good starting point for an API-based batch upload, where similarly assets are uploaded together with their metadata in a single request.

**Comment by awead** [comment]
Uploads are already atomic. They occur asynchronously in the upload page. After they've uploaded, the batch update page allows you to apply metadata to the group of files you just uploaded.

I would suggest putting any required fields such as document type in the upload form. That way, if the batch update is never executed, the document type and other fields, would be applied at the time the files are uploaded.

**Comment by scossu** [comment]
I like this approach. We should then move some mandatory fields to step 1:
- Document type
- Relationship (not mandatory, but if set, it has to be present as soon as the resource is persisted in Fedora)

This way step 2 is intended as an optional batch edit. 

We can discuss the implementation of this change and create a new ticket accordingly.

**Comment by scossu** [comment]
Migrated to issue tracking system.
